src/main/java/com/github/osisoft/ocs_sample_library_preview/BaseClient.java

```diff
@@ -148,6 +167,33 @@ public HttpURLConnection getConnection(URL url, String method) {
         return urlConnection;
     }
 
+    /**
+     * Helper function to get standard request headers
+     * 
+     * @return Map<String,String> object with http headers
+     */
+    public Map<String, String> getHttpHeadersForRequest() {
+        Map<String, String> headers = new Hashtable<String, String>();
+        headers.put("Accept", "*/*; q=1");
+        headers.put("Accept-Encoding", "gzip");
+        headers.put("Content-Type", "application/json");
+
+        return headers;
+    } 
+
+    /**
+     * Helper function to get request headers for communities APIs
+     * 
+     * @param communityId id of the community
+     * @return Map<String,String> object with http headers
+     */
+    public Map<String, String> getHttpHeadersForCommunitiesRequest(String communityId) {
+        Map<String, String> headers = getHttpHeadersForRequest();
+        headers.put("Community-id", communityId);
+
+        return headers;
+    } 
```

Comment:
I'm trying to think of the best way to implement this where wanting a community id header doesnt force you to use the underlying getHttpHeadersForRequest as is. 

Maybe having a second overload function that also accepts a Map<String, String> headers, and instead of calling the getHttpHeadersForRequest() and putting the community id header, it just puts the community id header into the passed headers, and returns it back. 

This is fine when community id is the only 'custom' header being used, but the approach quickly gets tedious when there's a possibility of multiple 'custom' headers being used.

Reply:
I don't think it needs to go quite to the extreme of the python library's 'additional_headers' approach because that would get messy with the headers and additional_headers having the same type, and java not supporting named arguments. It's probably easiest to just do a quick wrapper like this and call it done

Reply:
Yeah that is a limitation with this... Since implementation of this is so much more awkward in Java I am almost leaning towards keeping this simple solution for now and dealing with a more general, perhaps more complicated, solution should we add more headers in the future, but I am off course open to pursuing that now if we feel it's worth the effort.

That said, I could refactor it to accept a headers Map and just add the community header to better fit the case of more headers being added if you would prefer that, I don't have a strong opinion there.

Reply:
let's keep this one as is, and then on the next PR when I implement the verbosity header I'll refactor it a tiny bit as necessary